dbsnp-importer/src/test/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/RenormalizationProcessorTest.java

```diff
@@ -87,31 +107,133 @@ private int computeEnd(int position, String reference, String alternate) {
+        assertEquals(variant.getType(), renormalized.getType());
+
+        assertEquals(variant.getSourceEntries().size(), renormalized.getSourceEntries().size());
+        Iterator<VariantSourceEntry> variantSourceEntryIterator = variant.getSourceEntries().iterator();
+        Iterator<? extends IVariantSourceEntry> renormalizedSourceEntryIterator = renormalized.getSourceEntries().iterator();
+
+        while (variantSourceEntryIterator.hasNext() && renormalizedSourceEntryIterator.hasNext()) {
+            VariantSourceEntry sourceEntry = variantSourceEntryIterator.next();
+            IVariantSourceEntry renormalizedSourceEntry = renormalizedSourceEntryIterator.next();
+            assertEquals(sourceEntry.getStudyId(), renormalizedSourceEntry.getStudyId());
+            assertEquals(sourceEntry.getFileId(), renormalizedSourceEntry.getFileId());
+            assertEquals(sourceEntry.getFormat(), renormalizedSourceEntry.getFormat());
+            assertVariantStatisticsEquals(sourceEntry, renormalizedSourceEntry, reference, renormalizedAlternate);
+            assertEquals(sourceEntry.getSamplesData(), renormalizedSourceEntry.getSamplesData());
+            assertAttributesEquals(sourceEntry, renormalizedSourceEntry, variant);
+
+            // TODO jmmut: should this change?
+            assertArrayEquals(sourceEntry.getSecondaryAlternates(), renormalizedSourceEntry.getSecondaryAlternates());
+        }
+    }
+
+    private void assertVariantStatisticsEquals(VariantSourceEntry sourceEntry,
+                                               IVariantSourceEntry renormalizedSourceEntry,
+                                               String renormalizedReference,
+                                               String renormalizedAlternate) {
+        assertEquals(sourceEntry.getCohortStats().size(), renormalizedSourceEntry.getCohortStats().size());
+        Iterator<Map.Entry<String, VariantStatistics>> statsIterator =
+                sourceEntry.getCohortStats().entrySet().iterator();
+        Iterator<Map.Entry<String, VariantStatistics>> renormalizedStatsIterator =
+                renormalizedSourceEntry.getCohortStats().entrySet().iterator();
+        while (statsIterator.hasNext() && renormalizedStatsIterator.hasNext()) {
+            Map.Entry<String, VariantStatistics> statsEntry = statsIterator.next();
+            Map.Entry<String, VariantStatistics> renormalizedStatsEntry = renormalizedStatsIterator.next();
+            assertEquals(statsEntry.getKey(), renormalizedStatsEntry.getKey());
+            VariantStatistics renormalizedStats = renormalizedStatsEntry.getValue();
+            assertEquals(renormalizedReference, renormalizedStats.getRefAllele());
+            assertEquals(renormalizedAlternate, renormalizedStats.getAltAllele());
+            assertEquals(renormalizedAlternate, renormalizedStats.getMafAllele());
+            VariantStatistics stats = statsEntry.getValue();
```

Comment:
Can't we just compare equality of the `VariantStatistics`, `VariantSourceEntry`, etc, objects?

Reply:
No, it checks that the ref, alt and mafAllele are the renormalized ones. Probably the methods shouldn't be called assertEquals.